src/binder/clrprivbindercoreclr.cpp

```diff
@@ -159,8 +159,13 @@ HRESULT CLRPrivBinderCoreCLR::BindUsingPEImage( /* in */ PEImage *pPEImage,
                 {
                     if (pCoreCLRFoundAssembly->GetIsInGAC())
                     {
-                        // If we were able to bind to a TPA assembly, then fail the load
-                        IF_FAIL_GO(HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+                        SString tpaAssemblyPath;
+                        SString assemblyPath;
+                        BinderGetImagePath(pPEImage, assemblyPath);
+                        BinderGetImagePath(pCoreCLRFoundAssembly->GetPEImage(), tpaAssemblyPath);
+                        if(!tpaAssemblyPath.EqualsCaseInsensitive(assemblyPath))
```

Comment:
You should be able to just compare `PEImage` pointers. PEImages are in a hashtable with path as the key (check `PEImage::OpenImage`) so you should get the same one for same path.